pkg/route/api/types.go

```diff
@@ -89,11 +89,19 @@ type TLSConfig struct {
 	// DestinationCACertificate provides the contents of the ca certificate of the final destination.  When using reencrypt
 	// termination this file should be provided in order to have routers use it for health checks on the secure connection
 	DestinationCACertificate string `json:"destinationCACertificate,omitempty"`
+
+	// Insecure indicates the desired behavior for insecure connections
+	// to an edge-terminated route: expose, redirect or disable.
+	Insecure TLSInsecureType `json:"insecure,omitempty"`
```

Comment:
`InsecureEdgeTerminationPolicy` and `insecureEdgeTerminationPolicy` - I'd rather be a bit more explanatory